### PR TITLE
support range request for s3 SA

### PR DESCRIPTION
**Pull request by cherbst**

### CODE REVIEW

**Comment by cherbst** [comment]
I added a need null check for the option argument.

File: packages/s3/s3.server.js [hunk 1]
```diff
@@ -122,11 +122,15 @@ FS.Store.S3 = function(name, options) {
...
     createReadStream: function(fileKey, options) {
-
-      return S3.createReadStream({
+      var params = {
         Bucket: bucket,
         Key: folder + fileKey
-      });
+      };
+
+      if (options && options.start && options.end)
```
Comment (aldeed) [inline packages/s3/s3.server.js]: This condition will fail when start === 0 (unless start and end come in as strings, I can't remember)

**Comment by aldeed** [comment]
@cherbst, see my one comment. Thanks for the PR, though! cc @raix

**Comment by cherbst** [comment]
Good point, although I think range requests starting at byte 0 are quite uncommon, it should definitely be supported. Should I append to this PR or just change the commit? I am not sure if your comment gets lost then.

**Comment by aldeed** [comment]
@cherbst, just commit the fix to your s3-range-request branch so that this PR gets updated.

**Comment by cherbst** [comment]
Changed it.

**Pull request merged**